subprojects/common/core/src/main/java/hu/bme/mit/theta/core/type/functype/FuncExprs.java

```diff
@@ -46,8 +47,8 @@ public static <ParamType extends Type, ResultType extends Type> FuncAppExpr<Para
     private static <ParamType extends Type, ResultType extends Type> FuncAppExpr<ParamType, ResultType> App(
             final Expr<FuncType<ParamType, ResultType>> func, final Expr<ParamType> paramHead, final List<Expr> paramTail) {
         if (!paramTail.isEmpty()) {
-            final var newParamHead = paramTail.get(paramTail.size() - 1);
-            final var newParamTail = paramTail.subList(0, paramTail.size() - 1);
+            final var newParamHead = paramTail.get(0);
+            final var newParamTail = paramTail.subList(1, paramTail.size());
```

Comment:
Is this change intentional?

Reply:
Yes, unfortunately beforehand we mixed up parameters when transforming HOL two-argument predicates to multi-parameter FOL ones. Now a `(func a b)` is equivalent to a `((func a) b)` (before, it was `((func b) a)`).